### PR TITLE
Add new items in scan context 

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: ref #5926

Problem Summary:

### What is changed and how it works?

Add more information in Scan Context:
1. time cost of wait index 
2. read size from s3 and cache

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/S3/S3RandomAccessFile.cpp [hunk 1]
```diff
@@ -224,19 +224,23 @@ inline static RandomAccessFilePtr tryOpenCachedFile(const String & remote_fname,
...
+    if (scan_context.has_value())
+        scan_context.value()->total_s3_read_dmfile_size += filesize.value();
     auto & ins = S3::ClientFactory::instance();
     return std::make_shared<S3RandomAccessFile>(ins.sharedTiFlashClient(), remote_fname);
 }
 
 RandomAccessFilePtr S3RandomAccessFile::create(const String & remote_fname)
 {
-    return createFromNormalFile(remote_fname, read_file_info ? std::optional<UInt64>(read_file_info->size) : std::nullopt);
+    return createFromNormalFile(remote_fname, read_file_info ? std::optional<UInt64>(read_file_info->size) : std::nullopt, read_file_info ? std::optional<DM::ScanContextPtr>(read_file_info->scan_context) : std::nullopt);
```
Comment (JaySon-Huang) [inline dbms/src/Storages/S3/S3RandomAccessFile.cpp:244]: Can we ensure `read_file_info->scan_context` is not `nullptr`? If not, we should handle it properly.

```
auto scan_context = std::nullopt;
if (read_file_info && read_file_info->scan_context != nullptr)
{
  scan_context = std::optional<DM::ScanContextPtr>(read_file_info->scan_context);
}
```

**Comment by JaySon-Huang** [comment]
And there are some ColumnFile in the DeltaValueSpace read from local cache or fetch from remote write nodes. We should add them to `total_disagg_read_cache_hit_size`/`total_disagg_read_cache_miss_size`

https://github.com/pingcap/tiflash/blob/96fbffd914ea6ebe42ec3050d8930ec3873c4f64/dbms/src/Storages/DeltaMerge/Remote/RNWorkerFetchPages.cpp#L41-L71

File: dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.h [hunk 2]
```diff
@@ -329,7 +330,7 @@ class DMFilePackFilter
             return;
 
         Stopwatch watch;
-        loadIndex(param.indexes, dmfile, file_provider, index_cache, set_cache_if_miss, col_id, read_limiter);
+        loadIndex(param.indexes, dmfile, file_provider, index_cache, set_cache_if_miss, col_id, read_limiter, scan_context);
 
         scan_context->total_dmfile_rough_set_index_load_time_ns += watch.elapsed();
```
Comment (Lloyd-Pottiger) [inline dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.h:335]: add rs index check time into `total_dmfile_rough_set_index_load_time_ns` ?

Comment (hongyunyan) [reply dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.h:331]: done

File: dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.h [hunk 3]
```diff
@@ -193,8 +193,10 @@ class DMFilePackFilter
                 tryLoadIndex(attr.col_id);
             }
 
+            Stopwatch watch;
             const auto check_results = filter->roughCheck(0, pack_count, param);
             std::transform(use_packs.begin(), use_packs.end(), check_results.begin(), use_packs.begin(), [](UInt8 a, RSResult b) { return (static_cast<bool>(a)) && (b != None); });
+            scan_context->total_dmfile_rough_set_index_load_time_ns += watch.elapsed();
```
Comment (Lloyd-Pottiger) [inline dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.h:199]: rename to `total_dmfile_rough_set_index_check_time_ns` ?

Comment (JaySon-Huang) [reply dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.h:199]: yes, we can distinguish rough set index `loading time` and `checking time`

**Review by JaySon-Huang — COMMENTED** [review]
The rest LGTM

File: dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.h (hunk 3 above)
Comment (hongyunyan) [reply dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.h:199]: Does this mean that we need to use two indicators separately for statistics?

Comment (Lloyd-Pottiger) [reply dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.h:199]: I think the sum is enough.

**Review by JaySon-Huang — APPROVED** [review]

**Review by Lloyd-Pottiger — APPROVED** [review]

**Pull request merged**